src/kubernetes_api_objects/resource.rs

```diff
@@ -79,4 +79,67 @@ pub trait ResourceView: Sized {
                     ==> d.kind == Self::kind();
 }
 
+pub trait CustomResourceView: Sized {
```

Comment:
Seems that CustomResourceView repeats a lot of things inside ResourceView. Why not we merge the two rule methods into ResourceView (and not keep CustomResourceView)? For all the built-in types, their rules always return true anyway. This reduces the complexity in terms of traits we need to manage

Reply:
I'd rather wait for verus to support the check for supertrait.

Reply:
I am not sure whether supertrait is on their roadmap. The problem with this design is that it prevents us from implementing controllers that manage built-in resources, but we might not get there.

Reply:
You mean in that case the CR type is some built-in type?

Reply:
Another alternative is to wrap the validation related methods into a trait (maybe `Validation`?) and each concrete cr type implement `ResourceView + Validation`

Reply:
> You mean in that case the CR type is some built-in type?

Yes. People usually define their own CR type for controllers but sometimes they just manage the built-in types.